ambry-cloud/src/main/java/com/github/ambry/cloud/CloudUpdateValidator.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud;
+
+import com.github.ambry.store.StoreException;
+import com.github.ambry.store.StoreKey;
+import java.util.Map;
+
+
+/**
+ * Interface for Ambry validation logic for updates requested from cloud destination.
+ */
+public interface CloudUpdateValidator {
+  /**
+   * Validate operation on {@link CloudBlobMetadata} in cloud destination for the operation on blob with
+   * given {@link StoreKey} and new requested life version.
```

Comment:
new requested life version doesn't appear to be an argument.

Reply:
I'd like to see clearer description of what is validated, or what would cause validation to fail.  And specifically that you are validating application of update fields to existing metadata.

Reply:
Added comments to capture the description, while keeping it relatively generic so that it can potentially be applied to future CloudDestination extensions. Let me know if it looks good.